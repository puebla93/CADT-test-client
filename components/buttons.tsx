import styled from 'styled-components';


const ButtonsContainer = styled.div`
    display: flex;
    justify-content: center;
    margin: 48px 0;
`;


const Button = styled.button`
    background-color: ${({disabled}) => (disabled ? '#6b6b6b' : '#00a8b5')};
    border-radius: 24px;
    height: 48px;
    line-height: 24px;
    font-size: 16px;
    font-weight: bold;
    letter-spacing: normal;
    padding: 12px 24px;
    border: none;
    text-align: center;
    min-width: 80px;
    cursor: ${({disabled}) => (disabled ? 'initial' : 'pointer')};
    margin-left: 24px;
    
`;


interface InputButtonProps {
    children: React.ReactNode;
    type: string;
    accept: string;
    onChange?: (event: any) => void;
}


const LabelButton = styled.label`
    background-color: #00a8b5;
    border-radius: 24px;
    height: 48px;
    line-height: 24px;
    font-size: 16px;
    font-weight: bold;
    letter-spacing: normal;
    padding: 12px 24px;
    margin-right: 24px;
    border: none;
    text-align: center;
    min-width: 80px;
    cursor: pointer;

    > input{display: none !important;}
`;


const InputButton = (props: InputButtonProps): JSX.Element => {
    const {
        type,
        accept,
        children,
        onChange,
    } = props;

    return (
        <LabelButton>
            <input type={type} onChange={onChange} accept={accept} />
            {children}
        </LabelButton>
    );
};


export { ButtonsContainer, Button, InputButton };
