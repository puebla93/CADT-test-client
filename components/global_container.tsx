import styled from 'styled-components';

const GlobalConatiner = styled.div`
    padding: 32px;
`;

export default GlobalConatiner;
