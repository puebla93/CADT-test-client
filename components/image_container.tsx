import styled from 'styled-components';

const ImageConatiner = styled.div`
    display: flex;
    justify-content: space-evenly;
    margin-top: 48px;
    > img {
        max-width: 45%;
        flex: 1;
    }
`;

export default ImageConatiner;
