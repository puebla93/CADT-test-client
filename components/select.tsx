import styled from 'styled-components';

const Select = styled.select`
    margin-top: 48px;
    display: inline-block;
    border-radius: 12px;
    box-sizing: border-box;
    padding: 0.5em 2em 0.5em 0.5em;
    border: 1px solid #eee;
    font: inherit;
    line-height: inherit;
    -webkit-appearance: none;
    -moz-appearance: none;
    -ms-appearance: none;
    appearance: none;
    background-repeat: no-repeat;
    background-image: linear-gradient(45deg, transparent 50%, currentColor 50%), linear-gradient(135deg, currentColor 50%, transparent 50%);
    background-position: right 15px top 1em, right 10px top 1em;
    background-size: 5px 5px, 5px 5px;
`;

export default Select;
