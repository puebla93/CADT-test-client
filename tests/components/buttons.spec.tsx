import { render } from '@testing-library/react'
import { ButtonsContainer, Button, InputButton } from '../../components/buttons'

describe('Button component tests', () => {
    const buttonText = 'Here goes some text';

    test('Button matches snapshot', async () => {
        const {container} = render(
            <Button>{buttonText}</Button>
        );
        expect(container).toMatchSnapshot();
    });

    test('InputButton matches snapshot', async () => {
        const {container} = render(
            <InputButton type='file' accept='image/png, image/jpeg'>{buttonText}</InputButton>
        );
        expect(container).toMatchSnapshot();
    });

    test('ButtonsContainer matches snapshot', async () => {
        const {container} = render(
            <ButtonsContainer></ButtonsContainer>
        );
        expect(container).toMatchSnapshot();
    });
})
