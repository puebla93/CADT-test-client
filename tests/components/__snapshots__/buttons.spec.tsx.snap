// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests Button matches snapshot 1`] = `
<div>
  <button
    class="sc-gswNZR bRnJVC"
  >
    Here goes some text
  </button>
</div>
`;

exports[`Button component tests ButtonsContainer matches snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe eZAWPk"
  />
</div>
`;

exports[`Button component tests InputButton matches snapshot 1`] = `
<div>
  <label
    class="sc-dkrFOg SnwvQ"
  >
    <input
      accept="image/png, image/jpeg"
      type="file"
    />
    Here goes some text
  </label>
</div>
`;
